lib/telemetry_poller/application.ex

```diff
@@ -0,0 +1,19 @@
+defmodule Telemetry.Poller.Application do
+  @moduledoc false
+
+  def start(_type, _args) do
+    import Supervisor.Spec, only: [worker: 2]
+
+    default_poller_opts =
+      Application.get_env(:telemetry_poller, :default, name: Telemetry.Poller.Default)
```

Comment:
I think we should use `Keyword.merge` because otherwise if somebody sets:

    config :telemetry_poller, :default, vm_measurements: :default

then we won't set the name.

Speaking about defaults, I would also default to `vm_measurements: :default`. :)

Reply:
@josevalim `vm_measurements: :default` is already handled by `start_link/2`.